query_optimizer/ExecutionGenerator.cpp

```diff
@@ -1527,5 +1527,11 @@ void ExecutionGenerator::convertTableGenerator(
   temporary_relation_info_vec_.emplace_back(tablegen_index, output_relation);
 }
 
+S::InsertDestination* ExecutionGenerator::addNewInsertDestinationToQueryContext() {
+  std::unique_ptr<S::InsertDestination> insert_destination_proto(query_context_proto_->add_insert_destinations());
+  insert_destination_proto->set_query_id(query_context_proto_->query_id());
```

Comment:
Since you have set `query_id` in `QueryContext` proto, I was wondering why you have to do so in `InsertDestination` proto as well.